### DISCUSSION TITLE
how to get full tweet text for rwClient.v2.search( )

### DISCUSSION THREAD

**Discussion by MarufRien1** [plhery/node-twitter-api-v2#269] [Q&A] [answered]
i am trying to get the full text of the tweets returned by rwClient.v2.search( ) function.
but every time the function is returning only the first few sentences of a tweet.

![image](https://user-images.githubusercontent.com/74663078/165033850-8777fef0-9b23-474a-9917-cb4cb99cd826.png)

how can I fix this problem

**Comment by alkihis** [comment, marked as answer]
it's because they're retweets, and retweets are truncated following the 140 chars convention.

You need to ask for retweeted status aside with your request:
```ts
const search = await client.v2.search('My search text', { expansions: ['referenced_tweets.id'], 'tweet.fields': ['referenced_tweets'] })

for await (const tweet of search) {
  // Get the retweet object obtained from `includes.tweets` in response object
  const retweet = search.includes.retweet(tweet)
  // If retweet object found, use its text (bc its a retweet), otherwise use original tweet text
  const fullText = retweet ? retweet.text : tweet.text

  // Use {fullText}
}
```

**Comment by MarufRien1** [comment]
Thank you 😊
